### PR TITLE
[Logos] Generate getters & setters for %property even if it is never accessed

### PR DESCRIPTION
**Pull request by eswick**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Fixes an issue where `%property` getters & setters were removed by Clang's optimizer if the property was never accessed.

Does this close any currently open issues?
------------------------------------------
No

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
`%property` is awesome, we should keep improving it.

Where has this been tested?
---------------------------
**Operating System:** OS X 10.11

**Platform:** Uh...

**Target Platform:** iOS

**Toolchain Version:** Apple LLVM 7.0.2

**SDK Version:** iOS 9.2

### CODE REVIEW

**Comment by kirb** [comment]
Cheers, actually this explains why I was having a strange issue I couldn’t understand a few weeks back. :p 

Two issues:

* You’ve changed the indentation from tabs to spaces on some lines you modified.
* You should rebase to change the commit messages to have a `[logos]` prefix.

(Ignore the CI check for now; it’s currently broken.)

File: bin/logos.pl [hunk 1]
```diff
@@ -566,6 +566,8 @@ package main;
 		} elsif($line =~ /\G%property\s*(?:\((\s*\w+\s*(?:,\s*(?:\w|\=|:)+\s*)*)\))?\s*((?:\w+\s+\**)+)(\w+)\s*;/gc){
 			nestingMustContain($lineno, "%property", \@nestingstack, "hook", "subclass");
 
+			$currentClass->hasinstancehooks(1);
```
Comment (uroboro) [inline bin/logos.pl]: Is this line required if the edit on line 101 in bin/lib/Logos/Class.pm is in place?

**Comment by uroboro** [comment]
Are lines 29 and 106 in bin/lib/Logos/Generator/Base/Property.pm remnants of attempt to "fix" the unused function warnings? (`__attribute__((used))`)

And regarding the `\n` characters, I think we inline everything to not modify the line numbers of the unprocessed code.

**Comment by eswick** [comment]
Lines 29 and 106 aren't remnants; they're how the compiler issue is fixed.

I will fix the spaces (stupid Xcode), rebase the commit messages, and see if I can do everything on one line.

File: bin/logos.pl (hunk 1 above)
Comment (eswick) [inline bin/logos.pl]: Yes, it is. Can't remember exactly why but I know it *is* required.

**Comment by eswick** [comment]
Okay guys, I think that's everything. Able to merge?

**Pull request merged**